### PR TITLE
Fix 'EventTarget.prototype.addEventListener is not overridden' (close #1902)

### LINKED ISSUE #1902: EventTarget.prototype.addEventListener is not overridden
### What is your Test Scenario?
I'd like to test, whether screen size is bigger then 800px;
 
### What is the Current behavior?
It works properly in regular browser (chrome without testcafe), but it does not when running testcafe test. 
 
### What is the Expected behavior?
My code should run consistently both in testcafe and in regular browser
 
### What is your web application and your TestCafe test code?
https://github.com/bbonczek/testcafe-change-detection-issue

### Steps to Reproduce:
You need to compare this app in regular browser and in testcafe browser. Clone repository and run 
`
npm run install
`

Next thing you want to do is run application in normal browser. You can do it by running 
`
npm run start
`
and navigating to localhost:4200

try to resize window, to change it's height - you should see value switching beetween true and false:

working case in chrome browser

You can see mp4 version here: https://ezgif.com/video-to-gif/ezgif-2-271f5333a3b8.mp4

Now let's try to do the same, but in testcafe test. Don't stop dev server. Open other console and run 
`npm run testcafe`

Test will go into debug just after page loads. Click `Unlock Page` and try resizing again- this time nothing will happen:
not working case in testcafe chrome browser
mp4 version is here: 
https://ezgif.com/video-to-gif/ezgif-2-37d0daf5224f.mp4

### Your Environment details:
 
* testcafe version:                   0.23.3
* node.js version:                    v10.14.1
* command-line arguments:     testcafe chrome e2e/testcafe-buggy-test.test.js
* browser name and version:   Chrome 71.0.3578.98
* platform and version:           Windows 10

### PR DESCRIPTION
**Pull request by Farfurix**
Fix #1902 

### Changes
1. `EventTarget` native methods (non-IE11 case) were added to `nativeMethods.addEventListener`, `nativeMethods.removeEventListener` and `nativeMethods.dispatchEvent` methods.
2. `_createElementOverridedMethods` and `_createDocumentBodyOverridedMethods ` are combined to `createOverriddenMethods`.
3. `_getBodyEventListenerWrapper ` and `_getEventListenerWrapper` are combined to `_getEventListenerWrapper`.
4. The ["SVGElement.dispatchEvent should be overriden"](https://github.com/DevExpress/testcafe-hammerhead/pull/2203/files#diff-d08ffa5e7bead4978f8ff10e1f95c97c) test was fixed (SVG element creation).
5. Naming.
a) Methods: `...overrided...` -> `...overridden...`.
b) `eventListeningInfo` -> `eventCtx`.

### Native behavior checking (non IE11)

<details>
<summary>html:</summary>

```html
<html>
<head></head>
<body>
<script>
    var iframe = document.createElement('iframe');

    iframe.id = 'id1';

    iframe.onload = function () {
        console.log('> EventTarget.prototype.addEventListener = \'add\';');
        console.log('> EventTarget.prototype.removeEventListener = \'remove\';');
        console.log('> EventTarget.prototype.dispatchEvent = \'dispatch\';');
        EventTarget.prototype.addEventListener = 'add';
        EventTarget.prototype.removeEventListener = 'remove';
        EventTarget.prototype.dispatchEvent = 'dispatch';
        var xhr = new XMLHttpRequest();

        console.log('\n');

        console.log('window.addEventListener === \'add\':', window.addEventListener === 'add'); // "true"
        console.log('document.addEventListener === \'add\':', document.addEventListener === 'add'); // "true"
        console.log('document.body.addEventListener === \'add\':', document.body.addEventListener === 'add'); // "true"
        console.log('iframe.addEventListener === \'add\':', iframe.addEventListener === 'add'); // "true"
        console.log('\n');
        console.log('window.removeEventListener === \'remove\':', window.removeEventListener === 'remove'); // "true"
        console.log('document.removeEventListener === \'remove\':', document.removeEventListener === 'remove'); // "true"
        console.log('document.body.removeEventListener === \'remove\':', document.body.removeEventListener === 'remove'); // "true"
        console.log('iframe.removeEventListener === \'remove\':', iframe.removeEventListener === 'remove'); // "true"
        console.log('\n');
        console.log('window.dispatchEvent === \'dispatch\':', window.dispatchEvent === 'dispatch'); // "true"
        console.log('document.dispatchEvent === \'dispatch\':', document.dispatchEvent === 'dispatch'); // "true"
        console.log('document.body.dispatchEvent === \'dispatch\':', document.body.dispatchEvent === 'dispatch'); // "true"
        console.log('iframe.dispatchEvent === \'dispatch\':', iframe.dispatchEvent === 'dispatch'); // "true"
        console.log('\n');
        console.log('xhr.addEventListener === \'add\':', xhr.addEventListener === 'add');
        console.log('xhr.removeEventListener === \'remove\':', xhr.removeEventListener === 'remove');
        console.log('xhr.dispatchEvent === \'dispatch\':', xhr.dispatchEvent === 'dispatch');


        console.log('\n\n\n\n');
        console.log('> iframe.contentWindow.EventTarget.prototype.addEventListener = \'iframeAdd\';');
        console.log('> iframe.contentWindow.EventTarget.prototype.removeEventListener = \'iframeRemove\';');
        console.log('> iframe.contentWindow.EventTarget.prototype.dispatchEvent = \'iframeDispatch\';');
        iframe.contentWindow.EventTarget.prototype.addEventListener = 'iframeAdd';
        iframe.contentWindow.EventTarget.prototype.removeEventListener = 'iframeRemove';
        iframe.contentWindow.EventTarget.prototype.dispatchEvent = 'iframeDispatch';
        var iframeXhr = new iframe.contentWindow.XMLHttpRequest();

        console.log('\n');

        console.log('window.addEventListener === \'add\':', window.addEventListener === 'add'); // "true"
        console.log('iframe.addEventListener === \'add\':', iframe.addEventListener === 'add'); // "true"
        console.log('iframe.contentWindow.EventTarget.prototype.addEventListener === \'iframeAdd\': ', iframe.contentWindow.EventTarget.prototype.addEventListener === 'iframeAdd'); // "true"
        console.log('iframe.contentWindow.addEventListener === \'iframeAdd\': ', iframe.contentWindow.addEventListener === 'iframeAdd'); // "true"
        console.log('iframe.contentWindow.document.addEventListener === \'iframeAdd\': ', iframe.contentWindow.document.addEventListener === 'iframeAdd'); // "true"
        console.log('iframe.contentWindow.document.body.addEventListener === \'iframeAdd\': ', iframe.contentWindow.document.body.addEventListener === 'iframeAdd'); // "true"
        console.log('\n');
        console.log('window.removeEventListener === \'remove\':', window.removeEventListener === 'remove'); // "true"
        console.log('iframe.removeEventListener === \'remove\':', iframe.removeEventListener === 'remove'); // "true"
        console.log('iframe.contentWindow.EventTarget.prototype.removeEventListener === \'iframeRemove\': ', iframe.contentWindow.EventTarget.prototype.removeEventListener === 'iframeRemove'); // "true"
        console.log('iframe.contentWindow.removeEventListener === \'iframeRemove\': ', iframe.contentWindow.removeEventListener === 'iframeRemove'); // "true"
        console.log('iframe.contentWindow.document.removeEventListener === \'iframeRemove\': ', iframe.contentWindow.document.removeEventListener === 'iframeRemove'); // "true"
        console.log('iframe.contentWindow.document.body.removeEventListener === \'iframeRemove\': ', iframe.contentWindow.document.body.removeEventListener === 'iframeRemove'); // "true"
        console.log('\n');
        console.log('window.dispatchEvent === \'dispatch\':', window.dispatchEvent === 'dispatch'); // "true"
        console.log('iframe.dispatchEvent === \'dispatch\':', iframe.dispatchEvent === 'dispatch'); // "true"
        console.log('iframe.contentWindow.EventTarget.prototype.dispatchEvent === \'iframeDispatch\': ', iframe.contentWindow.EventTarget.prototype.dispatchEvent === 'iframeDispatch'); // "true"
        console.log('iframe.contentWindow.dispatchEvent === \'iframeDispatch\': ', iframe.contentWindow.dispatchEvent === 'iframeDispatch'); // "true"
        console.log('iframe.contentWindow.document.dispatchEvent === \'iframeDispatch\': ', iframe.contentWindow.document.dispatchEvent === 'iframeDispatch'); // "true"
        console.log('iframe.contentWindow.document.body.dispatchEvent === \'iframeDispatch\': ', iframe.contentWindow.document.body.dispatchEvent === 'iframeDispatch'); // "true"
        console.log('\n');
        console.log('iframeXhr.addEventListener === \'iframeAdd\':', iframeXhr.addEventListener === 'iframeAdd');
        console.log('iframeXhr.removeEventListener === \'iframeRemove\':', iframeXhr.removeEventListener === 'iframeRemove');
        console.log('iframeXhr.dispatchEvent === \'iframeDispatch\':', iframeXhr.dispatchEvent === 'iframeDispatch');
    };

    document.body.appendChild(iframe);
</script>
</body>
</html>

```
</details>

### Checked [user case](https://github.com/DevExpress/testcafe-hammerhead/issues/1902)
<details>
<summary>test.js</summary>

```js
import { Selector } from 'testcafe';

fixture `Fixture`
    .page `http://localhost:4200/`;

test('Resize page, to check whether scren width is bigger than 800px, or not', async t => {
    const result = Selector('body > app-root > h1');

    await t
        .maximizeWindow()
        .expect(result.innerText).eql('true');

    await t
        .resizeWindow(400, 600)
        .expect(result.innerText).eql('false');

    await t
        .resizeWindow(801, 1000)
        .expect(result.innerText).eql('true');
});

```
</details>

<details>
<summary>Result</summary>

```
 Running tests in:
 - Chrome 79.0.3945.130 / Windows 10

 Fixture
 √ Resize page, to check whether scren width is bigger than 800px, or not


 1 passed (5s)

```
</details>

### CODE REVIEW

File: src/client/sandbox/event/listeners.ts [hunk 1]
```diff
@@ -144,19 +144,19 @@ export default class Listeners extends EventEmitter {
         };
     }
 
-    private _createElementOverridedMethods (el: HTMLElement|Window|Document) {
+    createEventTargetOverriddenMethods () {
```
Comment (LavrovArtem) [inline src/client/sandbox/event/listeners.ts]: Why is this method public?

File: src/client/sandbox/event/listeners.ts [hunk 2]
```diff
@@ -144,19 +144,19 @@ export default class Listeners extends EventEmitter {
...
         const listeners = this;
 
         return {
             addEventListener: function (...args: any[]) {
                 const type                   = args[0];
                 const listener               = args[1];
```
Comment (LavrovArtem) [inline src/client/sandbox/event/listeners.ts]:
```js
const [type, listener] = args;
const el = this;
```

File: src/client/sandbox/event/listeners.ts [hunk 3]
```diff
@@ -235,18 +180,25 @@ export default class Listeners extends EventEmitter {
...
             },
-
             removeEventListener: function (...args: any[]) {
-                const type       = args[0];
-                const listener   = args[1];
-                const useCapture = Listeners._getUseCaptureParam(args[2]);
-
-                const eventListeningInfo = listeningCtx.getEventCtx(this, type);
+                const type                      = args[0];
+                const listener                  = args[1];
```
Comment (LavrovArtem) [inline src/client/sandbox/event/listeners.ts]: Same as in the `addEventListener` wrapper

File: src/client/sandbox/event/listeners.ts (hunk 1 above)
Comment (Farfurix) [reply src/client/sandbox/event/listeners.ts]: It is used [here](https://github.com/DevExpress/testcafe-hammerhead/pull/2203/files#diff-9185f923f93f18209c3f343f36ca2c54R628):
```ts
const overriddenMethods = this.listenersSandbox.createOverriddenMethods();
```

**Pull request merged**